File: src/domain/entities/bot.ts
export interface Bot {
  id: string;
  name: string;
  description: string | null;
  system_prompt: string | null;
  settings: Record<string, any>;
  is_active: boolean;
  tenant_id: string;
  created_at: string;
  updated_at: string;
}

export interface BotCreateInput {
  name: string;
  description?: string;
  system_prompt?: string;
  settings?: Record<string, any>;
}

export interface BotUpdateInput {
  name?: string;
  description?: string;
  system_prompt?: string;
  settings?: Record<string, any>;
  is_active?: boolean;
}

File: backend/app/schemas/bot.py
from pydantic import BaseModel, Field, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import Optional, Dict, Any

class BotCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    system_prompt: Optional[str] = None
    settings: Dict[str, Any] = Field(default_factory=dict)

class BotUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    system_prompt: Optional[str] = None
    settings: Optional[Dict[str, Any]] = None
    is_active: Optional[bool] = None

class BotOut(BaseModel):
    id: UUID
    name: str
    description: Optional[str]
    system_prompt: Optional[str]
    settings: Dict[str, Any]
    is_active: bool
    tenant_id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
